### PR TITLE
feat: allow redirecting all seasonal notifs to leagues url

### LINKED ISSUE #608: Add option for separate webhook for Leagues events
### Checklist

- [X] I've searched the [issues and pull requests](https://github.com/pajlads/Dinkplugin/issues?q=) for similar looking suggestions.
- [X] I've checked the `Unreleased` section of the [changelog](https://github.com/pajlads/DinkPlugin/blob/master/CHANGELOG.md) for newly added features that sound like my suggestion.

### Describe your Suggestion

When you meet the criteria for a webhook to be triggered for any existing category, check to see if the player is in Leagues and allow it to be sent to a separate webhook url override.

Examples:
- Loot notifications can go to a separate webhook if it's detected the player is playing Leagues
- CLog notifications can go to a separate webhook if it's detected the player is playing Leagues

### Reasoning

This would help allow Discord server admins to separate out the notifications coming from Leagues servers so that they aren't overwhelming existing notifications to help prevent main game notifications getting missed out on while Leagues is going on.

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #608

### CODE REVIEW

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 1]
```diff
@@ -403,17 +402,24 @@ default String metadataWebhook() {
...
+        name = "Seasonal Policy",
         description = "Whether to suppress notifications that occur on seasonal worlds like Leagues.<br/>" +
             "Note: the Leagues-specific notifier uses an independent config toggle",
         position = 1015,
         section = advancedSection
     )
-    default boolean ignoreSeasonal() {
-        return false;
+    default SeasonalPolicy seasonalPolicy() {
+        return SeasonalPolicy.ACCEPT;
```
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPluginConfig.java:413]: could also consider making this `FORWARD_TO_LEAGUES` by default (for new plugin installs)

File: src/main/java/dinkplugin/domain/SeasonalPolicy.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package dinkplugin.domain;
+
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public enum SeasonalPolicy {
+    ACCEPT("Notify normally"),
+    REJECT("Drop notification"),
+    FORWARD_TO_LEAGUES("Use Leagues URL");
```
Comment (Felanbird) [inline src/main/java/dinkplugin/domain/SeasonalPolicy.java:9]:
```suggestion
    ACCEPT("Notify normally"),
    FORWARD_TO_LEAGUES("Use Leagues URL");
    REJECT("Off"),
```
maybe this and change the description from `suppress` to `send`?

File: src/main/java/dinkplugin/DinkPluginConfig.java (hunk 1 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPluginConfig.java:413]: & when leagues override is not set, it just uses primary url

**Comment by iProdigy** [comment]
Tested the config migration, haven't tested the leagues forwarding

**Review by Felanbird — APPROVED** [review]
tested forward & off
image

**Review by pajlada — APPROVED** [review]

**Pull request merged**